src/libshadertrap/src/executor.cc

```diff
@@ -1201,41 +1203,171 @@ bool Executor::CheckEqualBuffers(CommandAssertEqual* assert_equal) {
+                  GL_ARRAY_BUFFER, 0, static_cast<GLsizeiptr>(buffer_size[index]),
+                  GL_MAP_READ_BIT));
+          if (mapped_buffer[index] == nullptr) {
+            GL_CHECKERR(&assert_equal->GetStartToken(), "glMapBufferRange");
+            return false;
+          }
+        }
+
+        for (size_t index = start_index; index < static_cast<size_t>(format_entry.count);index++) {
+          uint8_t value_1 =
+              mapped_buffer[0][index];  // NOLINT(clang-analyzer-core.NullDereference)
+          uint8_t value_2 =
+              mapped_buffer[1][index];  // NOLINT(clang-analyzer-core.NullDereference)
+          if (value_1 != value_2) {
+            std::stringstream stringstream;
+            stringstream << "Byte mismatch at index " << index << ": "
+                        << assert_equal->GetArgumentIdentifier1() << "[" << index
+                        << "] == " << static_cast<uint32_t>(value_1) << ", "
+                        << assert_equal->GetArgumentIdentifier2() << "[" << index
+                        << "] == " << static_cast<uint32_t>(value_2);
+            message_consumer_->Message(MessageConsumer::Severity::kError,
+                                      &assert_equal->GetStartToken(),
+                                      stringstream.str());
+            result = false;
+          }
+        }
+      }
+      start_index += format_entry.count;
+      break;
+      case CommandAssertEqual::FormatEntry::Kind::kFloat:
+      {
+  
+        float* mapped_buffer[2]{nullptr, nullptr};
+          for (auto index : {0, 1}) {
+            GL_SAFECALL(&assert_equal->GetStartToken(), glBindBuffer, GL_ARRAY_BUFFER,
+                        buffers[index]);
+            mapped_buffer[index] =
+                static_cast<float*>(gl_functions_->glMapBufferRange_(
+                    GL_ARRAY_BUFFER, 0, static_cast<GLsizeiptr>(buffer_size[index]),
+                    GL_MAP_READ_BIT));
+            if (mapped_buffer[index] == nullptr) {
+             GL_CHECKERR(&assert_equal->GetStartToken(), "glMapBufferRange");
+             return false;
+            }
+
+        }
+
+        for (size_t index = start_index; index < static_cast<size_t>(format_entry.count * sizeof(float)); index++) {
+          float value_1 =
+              mapped_buffer[0][index];  // NOLINT(clang-analyzer-core.NullDereference)
+          float value_2 =
+              mapped_buffer[1][index];  // NOLINT(clang-analyzer-core.NullDereference)
+          if (value_1 != value_2) {
+            std::stringstream stringstream;
+            stringstream << "Byte mismatch at index " << index << ": "
+                        << assert_equal->GetArgumentIdentifier1() << "[" << index
+                        << "] == " << static_cast<uint32_t>(value_1) << ", "
+                        << assert_equal->GetArgumentIdentifier2() << "[" << index
+                        << "] == " << static_cast<uint32_t>(value_2);
```

Comment:
Why do you cast `value_1` and `value_2` to uint32_t here? The whole point of your PR is that we want to be able to see float-level differences.

Reply:
Just for printing, but I agree that a cast to float is more suitable here. Done